src/System.Text.Json/tests/JsonNumberTests.cs

```diff
@@ -0,0 +1,651 @@
+                (JsonNumber number, out byte v) => number.TryGetByte(out v),
+                v => v);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-17)]
+        [InlineData(17)]
+        [InlineData(byte.MinValue - 1)]
+        [InlineData(byte.MaxValue + 1)]
+        [InlineData(short.MinValue)]
+        [InlineData(short.MaxValue)]
+        public static void TestShort(short value)
+        {
+            TestInitialization(
+                value,
+                v => new JsonNumber(v),
+                (number, v) => number.SetInt16(v),
+                number => number.GetInt16(),
+                (JsonNumber number, out short v) => number.TryGetInt16(out v),
+                v => v);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-17)]
+        [InlineData(17)]
+        [InlineData(0x2A)]
+        [InlineData(0b_0110_1010)]
+        [InlineData(short.MinValue - 1)]
+        [InlineData(short.MaxValue + 1)]
+        [InlineData(int.MinValue)]
+        [InlineData(int.MaxValue)]
+        public static void TestInt(int value)
+        {
+            TestInitialization(
+                value,
+                v => new JsonNumber(v),
+                (number, v) => number.SetInt32(v),
+                number => number.GetInt32(),
+                (JsonNumber number, out int v) => number.TryGetInt32(out v),
+                v => v);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-17)]
+        [InlineData(17)]
+        [InlineData((long)int.MinValue - 1)]
+        [InlineData((long)int.MaxValue + 1)]
+        [InlineData(long.MinValue)]
+        [InlineData(long.MaxValue)]
+        public static void TestLong(long value)
+        {
+            TestInitialization(
+                value,
+                v => new JsonNumber(v),
+                (number, v) => number.SetInt64(v),
+                number => number.GetInt64(),
+                (JsonNumber number, out long v) => number.TryGetInt64(out v),
+                v => v);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(-17)]
+        [InlineData(3.14)]
+        [InlineData(-15.5)]
+#if BUILDING_INBOX_LIBRARY
+        [InlineData(float.MinValue)]
+        [InlineData(float.MaxValue)]
+        [InlineData(ulong.MinValue)]
+        [InlineData(ulong.MaxValue)]
+#endif
+        public static void TestFloat(float value)
+        {
+            TestInitialization(
+                value,
+                v => new JsonNumber(v),
+                (number, v) => number.SetSingle(v),
+                number => number.GetSingle(),
+                (JsonNumber number, out float v) => number.TryGetSingle(out v),
+                v => v);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(-17)]
+        [InlineData(3.14)]
+        [InlineData(-15.5)]
+#if BUILDING_INBOX_LIBRARY
```

Comment:
When I run it locally for netfx ulong.Min works (because it's 0 and actually repeated test, so I will remove it at all), ulong.Max does not. Some of those throw `OverflowException` (double.X), some fail because of precision errors (`float.X`, `ulong.MaxValue`). But I am not sure if it make sense to include ifs in TestInitialization or write a new version of TestInitialization for just those few cases with AssertValue replaced wit AssertThrows. But case of double.MaxValue is actually tested in TestRationalGetMismatches, so I can add a platform-if there.